### PR TITLE
feat(v-hover): create new component

### PR DESCRIPTION
**Pull request by johnleider**
## Description
Created new `v-hover` component

## Motivation and Context
Allow the user to react to a hovered element with a wrapper

## How Has This Been Tested?
jest

## Markup:

<details>

```vue
<template>
  <v-app>
    <v-content class="pa-5">
      <v-hover close-delay="0" class="mb-5">
        <v-card
          slot-scope="{ hover }"
          :color="hover ? 'blue lighten-3' : 'blue lighten-4'"
          class="hide-overflow"
        >
          <v-card-text>Lorem Ipsum</v-card-text>
          <v-expand-transition>
            <div v-show="hover">
              <v-card-text style="height: 200px">Hidden Text</v-card-text>
            </div>
          </v-expand-transition>
        </v-card>
      </v-hover>
      <v-expansion-panel class="mb-5">
        <v-expansion-panel-content>
          <div slot="header">Title</div>
          <v-card height="300" color="purple"></v-card>
        </v-expansion-panel-content>
      </v-expansion-panel>

      <v-list>
        <v-list-group>
          <v-list-tile slot="activator">
            <v-list-tile-title>Title</v-list-tile-title>
          </v-list-tile>
          <v-list-tile @click="">
            <v-list-tile-title>List</v-list-tile-title>
          </v-list-tile>
        </v-list-group>
      </v-list>

<v-btn @click="isActive = !isActive">Click</v-btn>
    <v-hover v-model="isActive" disabled>
      <v-card
        :class="`elevation-${isActive ? 12 : 2}`"
      >
        <v-card-text>
          Lorem Ipsum
        </v-card-text>
      </v-card>
    </v-hover>
    </v-content>
  </v-app>
</template>
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes, `dev` for new features and breaking changes).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have created a PR in the [documentation](https://github.com/vuetifyjs/vuetifyjs.com) with the necessary changes.

### CODE REVIEW

File: test/unit/components/VHover/VHover.spec.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import Vue from 'vue'
+import { test } from '@/test'
+import VHover from '@/components/VHover'
+
+test('VForm.js', ({ mount }) => {
```
Comment (KaelWD) [inline test/unit/components/VHover/VHover.spec.js]: nice copy-paste

Comment (johnleider) [reply test/unit/components/VHover/VHover.spec.js]: 🤣

File: src/components/VHover/VHover.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import mixins from '../../util/mixins'
+import { consoleWarn } from '../../util/console'
+
+// Types
+import { VNode } from 'vue'
+
+export default mixins(
+  Delayable,
+  Toggleable
+).extend({
```
Comment (KaelWD) [inline src/components/VHover/VHover.ts]: Add `@vue/component` comment

File: src/components/VHover/VHover.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+      type: Boolean,
+      default: undefined
+    }
+  },
+
+  methods: {
+    __onMouseEnter () {
+      this.isActive = true
+    },
+    __onMouseLeave () {
```
Comment (KaelWD) [inline src/components/VHover/VHover.ts]: I used __ in VImg for render methods only, it doesn't really fit here

File: src/components/VHover/VHover.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  render (): any {
+    if (!this.$scopedSlots.default && this.value === undefined) {
+      consoleWarn('v-hover is missing a default scopedSlot or bound value', this)
+
+      return null
+    }
+
+    let element = null
+
+    // TODO: types are wrong - https://github.com/vuejs/vue/pull/8644
```
Comment (KaelWD) [inline src/components/VHover/VHover.ts]: Move TODO to line 48

File: src/stylus/generic/_transitions.styl [hunk 5]
```diff
@@ -107,17 +115,18 @@
     transform: translate(100%, 0)
 
 // Generic transitions
+.expand-transition
+  transition-default()
+
 .scale-transition
-  &-enter-active, &-leave-active
-    transition: $secondary-transition
+  transition-default()
```
Comment (KaelWD) [inline src/stylus/generic/_transitions.styl]: This was `$secondary-transition` transition before, dunno what the difference is.

File: src/components/VHover/VHover.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    if (!element || typeof element === 'string' || Array.isArray(element)) {
+      consoleWarn('v-hover should only contain a single element', this)
+
+      return element
+    }
+
+    if (!this.disabled) {
+      element.data!.on = element.data!.on || {}
+
+      element.data!.on!.mouseenter = () => {
```
Comment (KaelWD) [inline src/components/VHover/VHover.ts]: Does this overwrite existing listeners? Might need to merge them: https://github.com/vuejs/vue/blob/25a688ee314d82b584ef0f192d4016b75664a616/src/core/instance/render-helpers/bind-object-listeners.js#L13

**Pull request merged**